x/posts/internal/types/post.go

```diff
@@ -90,6 +90,17 @@ func (ids PostIDs) Equals(other PostIDs) bool {
 	return true
 }
 
+// AppendIfMissing appends the given post to the list of posts if it does not exist inside it yet,
+// returning also true if the object has been appended successfully.
+func (ids PostIDs) AppendIfMissing(id PostID) (PostIDs, bool) {
+	for _, ele := range ids {
+		if ele.Equals(id) {
+			return nil, false
```

Comment:
```suggestion
			return ids, false
```

I would suggest not to return nil, but to return the original slice instead. This would be more coherent with the method name which suggests that nothing is performed if the ID already exists.

Reply:
yes, you're right.